lib/Analysis/SeaBuiltinsInfo.cc

```diff
@@ -270,6 +336,21 @@ Function *SeaBuiltinsInfo::mkTrackingOffFn(Module &M) {
   return FN;
 }
 
+Function *SeaBuiltinsInfo::mkFreeFn(Module &M) {
+  auto &C = M.getContext();
+  auto FC = M.getOrInsertFunction(SEA_FREE, Type::getVoidTy(C),
+                                  Type::getInt8PtrTy(C));
+  auto *FN = dyn_cast<Function>(FC.getCallee());
+  if (FN) {
+    FN->setDoesNotThrow();
+    FN->setDoesNotRecurse();
+    FN->addParamAttr(0, Attribute::NoCapture);
```

Comment:
should also be marked with `doesNotFreeMemory()`. While you can the function `free`, a more appropriate name is `mark_free`. The function marks a memory region as freed, but it does not free the memory region itself. The freeing will be done by calling libc `free` or C++ `delete`.